libs/datasets/timeseries.py

```diff
@@ -1001,6 +1006,57 @@ def drop_new_case_outliers(
     return new_timeseries
 
 
+def drop_tail_positivity_outliers(
+    dataset: MultiRegionDataset,
+    zscore_threshold: float = 10.0,
+    diff_threshold_ratio: float = 0.015,
+) -> MultiRegionDataset:
+    """Identifies and drops outliers from the new case series.
+
+    Args:
+        dataset: Timeseries.
+        zscore_threshold: Z-score threshold.  All new cases with a zscore greater than the
+            threshold will be removed.
+        diff_threshold_ratio: Minimum difference required for value to be outlier.
+
+    Returns: timeseries with outliers removed from test_positivity_7d
+    """
+    df_copy = dataset.timeseries.copy()
```

Comment:
Add TODO about being more consistent in how this accesses the data so readers know to not copy this pattern.